server/src/shared/auth/user-auth-handler.ts

```diff
@@ -2,29 +2,40 @@ import express from 'express'
 import { Result } from '../core/result'
 import { LoginUserErrors } from '../../modules/users/application/use-cases/login-user/login-user-errors'
 import { AppError } from '../core/app-error'
-import '../../setup/http/express/auth'
+import { AuthCodeString } from '../../modules/users/domain/value-objects/auth-code'
 import { UserDTO } from '../../modules/users/mappers/user-dto'
 
 export type AuthToken = string
 //if we choose to adopt other authentication certificate types (like csrf-tokens), this can change
-export type AuthCertificate = AuthToken 
+export type AuthCertificate = AuthCodeString
 
 //establish success, error, and combined response types for all authentication methods
 
 //login
 export type UserAuthHandlerLoginSuccess = {
-  user: UserDTO,
-  authCert: AuthCertificate
+  cert?: AuthCertificate
+  user: UserDTO
 }
 export type UserAuthHandlerLoginError = LoginUserErrors.IncorrectPasswordError | AppError.UnexpectedError
 export type UserAuthHandlerLoginResponse = Result<UserAuthHandlerLoginSuccess, UserAuthHandlerLoginError> 
 
+export interface UserAuthHandlerLoginOptions {
+  req: express.Request, // we make the assumption that all auth handlers will require these
+  res: express.Response,
+  [key: string]: any //additional, implementation-specific options for auth handlers
+}
+
 //creation
 export type UserAuthHandlerCreateSuccess = AuthCertificate
-export type UserAuthHandlerCreateError = AppError.UnexpectedError
+export type UserAuthHandlerCreateError = Error
```

Comment:
Why is this just an `Error` now instead of `AppError.UnexpectedError`?

Reply:
Good call, swapped to the more specific Error subclass.